src/main/java/net/snowflake/client/jdbc/SnowflakeSQLLoggedException.java

```diff
@@ -236,17 +266,34 @@ public SnowflakeSQLLoggedException(SFSession session, ErrorCode errorCode, Objec
     String reason =
         errorResourceBundleManager.getLocalizedMessage(
             String.valueOf(errorCode.getMessageCode()), params);
-    sendTelemetryData(null, reason, null, -1, errorCode, session);
+    sendTelemetryData(null, reason, null, -1, errorCode, session, this);
   }
 
   public SnowflakeSQLLoggedException(SFSession session, SFException e) {
     super(e);
     // add telemetry
-    sendTelemetryData(null, null, null, -1, null, session);
+    sendTelemetryData(null, null, null, -1, null, session, this);
   }
 
   public SnowflakeSQLLoggedException(SFSession session, String reason) {
     super(reason);
-    sendTelemetryData(null, reason, null, -1, null, session);
+    sendTelemetryData(null, reason, null, -1, null, session, this);
+  }
+
+  /**
+   * Log all SQLFeatureNotSupported exceptions with this function. Has to be called before throwing
+   * exception. These contain no query ID or errorCodes or vendorCodes. We can add a SQLState.
+   *
+   * @param session
+   */
+  public static void logSqlFeatureNotSupportedException(SFSession session) {
```

Comment:
Create a static function that should be called before every throwing of a SqLFeatureNotSupportedException. Since SqLFeatureNotSupportedException is an entirely different type of exception than SQLException, it cannot be added to this class as a constructor.